pkg/client/unversioned/clientcmd/loader.go

```diff
@@ -332,7 +332,17 @@ func LoadFromFile(filename string) (*clientcmdapi.Config, error) {
 	// set LocationOfOrigin on every Cluster, User, and Context
 	for key, obj := range config.AuthInfos {
 		obj.LocationOfOrigin = filename
+		if len(obj.Token) == 0 && len(obj.TokenFile) > 0 {
```

Comment:
> by precedence rules the token will be "foo" while it should actually be the contents of "someFile"
> 
> So I think we need to do the token loading pre-merge.
> 
> (although the merge logic is a little opaque to me, so I could be wrong)

I think you're correct about the behavior, but I'd still want this to be consistent with all the other loading behavior.

I'm open to changing that behavior wholesale (you make a good point about unexpected precedence behavior), but I think that inconsistent behavior would be worse than weird behavior.

Can we make this consistent and then change them all together?